File: design-patterns/JavaDesignPatterns/src/com/bigboxcode/designpattern/factory/transportmethod/AirTransportFactory.java
package com.bigboxcode.designpattern.factory.transportmethod;

public class AirTransportFactory extends TransportFactory {
    @Override
    public Transport getTransport(String name) {
        if (name.equalsIgnoreCase("plane")) {
            return new Plane();
        }

        if (name.equalsIgnoreCase("helicopter")) {
            return new Helicopter();
        }

        return null;
    }
}

File: design-patterns/php/factory/transportmethod/AirTransportFactory.php
<?php
// AirTransportFactory.php

namespace BigBoxCode\DesignPattern\Factory\TransportMethod;

class AirTransportFactory extends TransportFactory {
    public function getTransport(string $name): ?Transport {
        switch (strtolower($name)) {
            case "plane":
                return new Plane();
            case "helicopter":
                return new Helicopter();
        }

        return null;
    }
}